internal/jasmine_node_test/jasmine_runner.js

```diff
@@ -11,45 +12,113 @@ const UTF8 = {
 const BAZEL_EXIT_TESTS_FAILED = 3;
 const BAZEL_EXIT_NO_TESTS_FOUND = 4;
 
+// ibazel will write this string after a successful build
+// We don't want to re-trigger tests if the compilation fails, so
+// we should only listen for this event.
+const IBAZEL_NOTIFY_BUILD_SUCCESS = 'IBAZEL_BUILD_COMPLETED SUCCESS';
+const IBAZEL_NOTIFY_CHANGES = 'IBAZEL_NOTIFY_CHANGES';
+
 // Set the StackTraceLimit to infinity. This will make stack capturing slower, but more useful.
 // Since we are running tests having proper stack traces is very useful and should be always set to
 // the maximum (See: https://nodejs.org/api/errors.html#errors_error_stacktracelimit)
 Error.stackTraceLimit = Infinity;
 
-function main(args) {
-  if (!args.length) {
-    throw new Error('Spec file manifest expected argument missing');
-  }
-  const manifest = require.resolve(args[0]);
-  // Remove the manifest, some tested code may process the argv.
-  process.argv.splice(2, 1)[0];
+function watchMode(manifestLocation) {
+  // ibazel communicates with us via stdin
+  const rl = createInterface({input: process.stdin, terminal: false});
+  rl.on('line', chunk => {
+    if (chunk === IBAZEL_NOTIFY_BUILD_SUCCESS) {
+      // TODO: we should try keep the same instance of JasmineRunner if it's expensive to new up
+      // TODO: stop previous runs if we get multiple calls of this in quick succession (debounce)
+      const jrunner = createRunner(true);
+      addFiles(manifestLocation, jrunner);
+      jrunner.execute();
+    }
+  });
+  rl.on('close', () => {
+    // Give ibazel 5s to kill our process, otherwise do it ourselves
+    setTimeout(() => {
+      console.error('ibazel failed to stop jasmine after 5s; probably a bug');
+      process.exit(1);
+    }, 5000);
+  });
+}
+
+function deleteFromCache(fileName) {
+  // jasmine uses `require(fileName)` to read the files into the program
+  // so we have to delete them from the cache to ensure they read again on subsequent runs
+  // when in ibazel mode
+  delete require.cache[require.resolve(fileName)];
+}
+
+function addFiles(manifestLocation, jrunner){
+  deleteFromCache(manifestLocation)
+
+  const manifest = require.resolve(manifestLocation);
+  const file = fs.readFileSync(manifest, UTF8);
+  fs.readFileSync(manifest, UTF8).split('\n').filter(l => l.length > 0).forEach(f => {
+    // ensure it's actually read from disk again
+    deleteFromCache(f);
+    try {
+      // try requiring it incase there are syntax errors that kills the program later on
+      require(f);
+      jrunner.addSpecFile(f);
+    } catch (error) {
+      // dont add the files to jasmine since it'll run them and it'll kill the process
```

Comment:
is this feature related to the watch mode? what happens with the code at HEAD when we run into some file that has syntax errors?
if it's orthogonal to the watch mode, let's move to different PR

Reply:
It's not directly related to watch mode, but you wouldn't see this specific error in any other scenario.
On HEAD the process only ever runes once, so it dosen't matter if jasmine tries using require since it fails the test anyway.